sql/core/src/test/scala/org/apache/spark/sql/execution/WholeStageCodegenSuite.scala

```diff
@@ -307,4 +309,47 @@ class WholeStageCodegenSuite extends QueryTest with SharedSQLContext {
       // a different query can result in codegen cache miss, that's by design
     }
   }
+
+  test("SPARK-23598: Avoid compilation error with a lot of aggregation operations") {
```

Comment:
It's not a compilation but a runtime error.  maybe better say: `codegen working for lots of aggregation operations without runtime errors`